Split current item display block into item provider and item display blocks.

HyperPigeon opened the pull request:
- Added item provider block
- Removed give item option from item display block.
- Added scale and xyz offset display options to item display block.
- Abstracted item display block.

HyperPigeon committed:
- Added item provider block
- abstracted item display block
- added scale and xyz offset to item display block

sleepingdragoninn reviewed (changes requested):
Thanks for this! Looking great, we super appreciate it.

Display Block Requests:
- Replace rotation type with manual pitch/yaw +- (basically, everything is now `locked`)
- Remove Offset (Z offset does this now)
- Remove show name (text blocks achieve this better)

Provider Requests
- Block name translation!
- Render the placeholder as a billboard (like popup blocks) when the item will do the same.
- Render the placeholder offset against the wall when the item will do the same.
- Force gives item on, remove the option (prevent the screen from opening but don't remove it entirely)

HyperPigeon committed:
Made requested changes.

sleepingdragoninn commented:
More stuff!
- Pitch and yaw have logical ranges, is a discrete slider simple enough to add? would also solve the issue in the screenshot for manual or old blocks.
- Pitch and yaw are displayed in something that isn't degrees, can we get them in degrees?
- Provider placeholder doesn't billboard when the item will (wall offset is great though!)
- Provider tooltip should now say "interact with a glowcase item to unset"
![image](https://github.com/user-attachments/assets/829d89f6-67a7-459e-8f56-5ad770694cc5)

HyperPigeon committed:
- Updated item provider tooltip
- Converted pitch and yaw display in item display edit screen to degrees
- Provider placeholder now billboards when the item will

HyperPigeon commented:
@sisby-folk Do you know any example sliders I can use as a reference?

sleepingdragoninn commented:
gui scale?

HyperPigeon committed:
- Increased pitch and yaw increment/decrement amount from 1 to 15.

sleepingdragoninn commented:
could you please
- remove references/changes to rotation and offset properties in item provider code
- Remove rotation, offset, display values from c2s edit provider
- Reinstate the edit screen so that providers can be set to "ONCE" again 
- Skip the "NO" option when cycling the gives item button

Thanks pigeon!

HyperPigeon committed:
- Renamed record
- Re-enabled menu for ItemProviderBlock

HyperPigeon committed:
- Removed unchanging values.

HyperPigeon committed:
Merge branch '1.21' into item-block-split

HyperPigeon committed:
- Fixed merge issues
